GeneratorInterface/RivetInterface/plugins/HTXSRivetProducer.cc

```diff
@@ -0,0 +1,257 @@
+/* \class HTXSRivetProducer
+ *
+ * \author David Sperka, University of Florida
+ *
+ * $Id: HTXSRivetProducer.cc,v 1.1 2016/09/27 13:07:29 dsperka Exp $
+ *
+ */
+
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/Run.h"
+
+#include "SimDataFormats/GeneratorProducts/interface/HepMCProduct.h"
+#include "SimDataFormats/GeneratorProducts/interface/LHERunInfoProduct.h"
+
+#include "Rivet/AnalysisHandler.hh"
+#include "GeneratorInterface/RivetInterface/src/HiggsTemplateCrossSections.cc"
+#include "SimDataFormats/HTXS/interface/HiggsTemplateCrossSections.h"
+
+#include <vector>
+#include <cstdio>
+#include <cstring>
+
+
+using namespace Rivet;
+using namespace edm;
+using namespace std;
+
+class HTXSRivetProducer : public edm::EDProducer {
+public:
+    
+    explicit HTXSRivetProducer(const edm::ParameterSet& cfg) : 
+        _hepmcCollection(consumes<HepMCProduct>(cfg.getParameter<edm::InputTag>("HepMCCollection"))),
+        _lheRunInfo(consumes<LHERunInfoProduct,edm::InRun>(cfg.getParameter<edm::InputTag>("LHERunInfo")))
+    {
+        _analysisHandler = new Rivet::AnalysisHandler();
```

Comment:
if these need to be pointers (doesn't seem so) - use smart pointers

Reply:
would something similar to
std::auto_ptr<Rivet::AnalysisHandler> _analysisHandler (new Rivet::AnalysisHandler);
(which doesn't compile) be ok?

Reply:
not auto_ptr - but right.std::unique_ptr<Rivet::AnalysisHandler> _analysisHandler(new Rivet::AnalysisHandler()

but likely easier if _analysisHandler was a River::AnalysisHandler() and not a pointer of any type..

Reply:
I tried to use unique_ptr without success...